core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilter.java

```diff
@@ -15,33 +15,231 @@
-
-import javax.servlet.FilterChain;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
+import org.springframework.web.util.UriComponentsBuilder;
 
 /**
  * @author Joe Grandja
+ * @author Paurav Munshi
+ * @since 0.0.1
  */
 public class OAuth2AuthorizationEndpointFilter extends OncePerRequestFilter {
-	private Converter<HttpServletRequest, OAuth2AuthorizationRequest> authorizationRequestConverter;
+
+	private static final String DEFAULT_ENDPOINT = "/oauth2/authorize";
+
+	private Converter<HttpServletRequest, OAuth2AuthorizationRequest> authorizationRequestConverter = new OAuth2AuthorizationRequestConverter();
 	private RegisteredClientRepository registeredClientRepository;
 	private OAuth2AuthorizationService authorizationService;
-	private StringKeyGenerator codeGenerator;
+	private StringKeyGenerator codeGenerator = new Base64StringKeyGenerator();
+	private RedirectStrategy authorizationRedirectStrategy = new DefaultRedirectStrategy();
+	private RequestMatcher authorizationEndpointMatcher = new AntPathRequestMatcher(DEFAULT_ENDPOINT);
+
+	public OAuth2AuthorizationEndpointFilter(RegisteredClientRepository registeredClientRepository,
+			OAuth2AuthorizationService authorizationService) {
+		Assert.notNull(registeredClientRepository, "registeredClientRepository cannot be null.");
+		Assert.notNull(authorizationService, "authorizationService cannot be null.");
+		this.registeredClientRepository = registeredClientRepository;
+		this.authorizationService = authorizationService;
+	}
+
+	public final void setAuthorizationRequestConverter(
+			Converter<HttpServletRequest, OAuth2AuthorizationRequest> authorizationRequestConverter) {
+		Assert.notNull(authorizationRequestConverter, "authorizationRequestConverter cannot be set to null");
+		this.authorizationRequestConverter = authorizationRequestConverter;
+	}
+
+	public final void setCodeGenerator(StringKeyGenerator codeGenerator) {
+		Assert.notNull(codeGenerator, "codeGenerator cannot be set to null");
+		this.codeGenerator = codeGenerator;
+	}
+
+	public final void setAuthorizationRedirectStrategy(RedirectStrategy authorizationRedirectStrategy) {
+		Assert.notNull(authorizationRedirectStrategy, "authorizationRedirectStrategy cannot be set to null");
+		this.authorizationRedirectStrategy = authorizationRedirectStrategy;
+	}
+
+	public final void setAuthorizationEndpointMatcher(RequestMatcher authorizationEndpointMatcher) {
+		Assert.notNull(authorizationEndpointMatcher, "authorizationEndpointMatcher cannot be set to null");
+		this.authorizationEndpointMatcher = authorizationEndpointMatcher;
+	}
+
+	@Override
+	protected boolean shouldNotFilter(HttpServletRequest request) throws ServletException {
+		boolean pathMatch = this.authorizationEndpointMatcher.matches(request);
+		String responseType = request.getParameter(OAuth2ParameterNames.RESPONSE_TYPE);
```

Comment:
According to the spec of [Error response](https://tools.ietf.org/html/rfc6749#section-4.1.2.1) for Authorization request, the server should respond with `invalid_request` if missing an required parameter or `unsupported_response_type` for an invalid response type. So ideally the check for `response_type` should be added at the place of validating the auth request.

Reply:
That was how it was. I, initially, had the request validation also perform response_type check. But @jgrandja suggested to perform this check at the time of verifying if filter is applicable or not. Please look at this [comment](https://github.com/spring-projects-experimental/spring-authorization-server/pull/77#discussion_r423099177). And I think this is also a good approach because `response_type `could also be `token ` and in that case a different `Filter` will handle the request and I think for erroneous `response_type` we can have an separate error filter. I am fine with both the options, but just wanted to point out that this was kept this way deliberately.